### PR TITLE
PEM-6153-fixed the fragment url to follow the actual content

### PR DESCRIPTION
**Pull request by nage1234**
## Describe the Change

The change made as part of this PR is, when we load the packs doc page with fragment identifier (followed by "#"), it takes to the exact fragment in the "Additional Details"

## Changed Pages

No access to netify, but, it doesnt impact on the design , in the packs page, go to pack detail page of any packs.
or directly hit the url,
http://localhost:3000/integrations/packs/?pack=kubernetes-tke&version=1.24.4&parent=1.24.x#change-cluster-dns-service-domain
the control goes to the fragement anchor change-cluster-dns-service-domain

## Jira Tickets

https://spectrocloud.atlassian.net/browse/PEM-6153

## Backports

Can this PR be backported?

- [ ] Yes. _Remember to add the relevant backport labels to your PR._
- [ ] No. _Please leave a short comment below about why this PR cannot be backported._

### CODE REVIEW

File: src/components/PacksReadme/PacksReadme.tsx [hunk 1]
```diff
@@ -218,6 +227,14 @@ export default function PacksReadme() {
       });
     }, []);
 
+    useEffect(() => {
+      if (document && fragmentIdentifier) {
+        const elementId = fragmentIdentifier.replace("#", "");
+        const parent = document.getElementById?.(elementId);
+        parent?.querySelector?.("a")?.click();
```
Comment (nageshsairam1610) [inline src/components/PacksReadme/PacksReadme.tsx:234]: Can we use `targetElement?.scrollIntoView();` if the primary purpose of the click is to scroll to the Terraform section?

Comment (nage1234) [reply src/components/PacksReadme/PacksReadme.tsx:234]: i was using scrollintoview before committing the code changes, but, it was not working.. So, i had to use click

Comment (nage1234) [reply src/components/PacksReadme/PacksReadme.tsx]: let me try again once

**Review by nageshsairam1610 — APPROVED** [review]
Looks good to me

File: src/components/PacksReadme/PacksReadme.tsx [hunk 2]
```diff
@@ -252,8 +253,29 @@ export default function PacksReadme() {
...
+      }
+    }, [fragmentIdentifier, componentsMounted]);
+
+    useEffect(() => {
+      if (packData) {
+        setTimeout(() => {
+          isComponentsMounted(true);
```
Comment (nageshsairam1610) [inline src/components/PacksReadme/PacksReadme.tsx:266]:
```useEffect(() => {
  if (isBrowser && fragmentIdentifier) {
    const timeoutId = setTimeout(() => {
      const elementId = fragmentIdentifier.replace("#", "");
      const parent = document.getElementById?.(elementId);
      parent?.querySelector?.("a")?.click();
    }, 1000);

    return () => clearTimeout(timeoutId);
  }
}, [fragmentIdentifier]);```

If this work. We can get rid of `componentsMounted`

**Review by nageshsairam1610 — COMMENTED** [review]
minor change

File: src/components/PacksReadme/PacksReadme.tsx (hunk 2 above)
Comment (nage1234) [reply src/components/PacksReadme/PacksReadme.tsx]: it works.. done! the state variable is not needed

**Review by nageshsairam1610 — APPROVED** [review]

**Review by karl-cardenas-coding — APPROVED** [review]
Thank you so much @nage1234 🙇🏻
